### PR TITLE
[JENKINS-58157] - Fix handling of long build names in Web UI

### PR DESCRIPTION
**Pull request by aakashkanvinde**
See [JENKINS-58157](https://issues.jenkins-ci.org/browse/JENKINS-58157).

### Proposed changelog entries

* Entry 1: Minor UI changes: 
When build name is too long, for 'Delete Build', 'Build-caption page-headline' and 'Breadcrumbs name', the overflowed content is hidden and ellipsis are displayed 

### Desired reviewers
@jenkinsci/jenkins

image

### CODE REVIEW

**Comment by aakashkanvinde** [comment]
@daniel-beck @oleg-nenashev can you please review this PR once?

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me,, but I am am not a web developer at all.
Better call Gavin (c)
@halkeye

File: core/src/main/resources/lib/layout/breadcrumbs.css
Comment (halkeye) [inline core/src/main/resources/lib/layout/breadcrumbs.css]: Minor nitpick but you have a rogue newline here.

File: war/src/main/webapp/css/responsive-grid.css
Comment (halkeye) [inline war/src/main/webapp/css/responsive-grid.css]: This does mean if they have a resolution higher than 1200px it'll still be cut off right?

**Review by halkeye — COMMENTED** [review]
I'm afraid I'm not super familiar with the CSS for Jenkins. You've made a lot of changes to a lot of classes which makes me a little nervous

Are all the class changes to fix it he link and the header?

**Comment by halkeye** [comment]
Generally looks good too me, Im just scared of unintended side effects.

File: core/src/main/resources/lib/layout/breadcrumbs.css
Comment (aakashkanvinde) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: hey @halkeye, I don't know how that newline has come up there. That is not showing on my local code. I think that can be ignored.

File: war/src/main/webapp/css/responsive-grid.css
Comment (aakashkanvinde) [reply war/src/main/webapp/css/responsive-grid.css]: Well, I have checked that on viewport sized up to 1900px and everything worked fine. I can't say for sure what might happen for viewport sizes above 1900px

**Comment by aakashkanvinde** [comment]
> Generally looks good too me, I'm just scared of unintended side effects.

@halkeye as far as my changes are concerned, there won't be any side effects as I have just focussed my changing the headline, breadcrumbs and button name length.

**Comment by halkeye** [comment]
I'm not overly concerned, just leaving notes for Oleg and the other core maintainers.

I think max width of 1900 is reasonable. And if I remember correctly just the breadcrumb not the entire page.

**Comment by aakashkanvinde** [comment]
> I'm not overly concerned, just leaving notes for Oleg and the other core maintainers.
> 
> I think max width of 1900 is reasonable. And if I remember correctly just the breadcrumb, not the entire page.

Ok, so are you ok with the changes and end result?

**Comment by aakashkanvinde** [comment]
> > I'm not overly concerned, just leaving notes for Oleg and the other core maintainers.
> > I think max width of 1900 is reasonable. And if I remember correctly just the breadcrumb, not the entire page.
> 
> So are you ok with the changes and end result and if yes can you please approve?

@oleg-nenashev do you have any suggestions on this?

**Comment by halkeye** [comment]
I'm super reluctant to put my mark of approval on something I havn't done a deep dive into but sure, its not any worse than before.

Oleg and the others are probably super busy with jenkins world coming up next week, so i wouldn't expect a whole lot for a bit.

**Review by halkeye — APPROVED** [review]

File: core/src/main/resources/lib/layout/breadcrumbs.css
Comment (halkeye) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: its probably newline related, file is windows and you are mac or other way around.

run the file through cat and see what it looks like.

not a blocker to me

Comment (aakashkanvinde) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: oh alright, should I commit the code again in this case?

Comment (halkeye) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: Lets go with yes. Better to clean up after yourself. You can probably just edit the code directly in github.

If everyone leaves one or two random extra lines behind, it can get messy, better to stay consistent.

**Comment by halkeye** [comment]
The build failure looks unrelated, but just in case I would trigger a rebuild.

Close the pr, wait like 30 seconds to a minute, then re-open the PR, it should trigger a rebuild that way.

✔ help reviewers know its ready

File: core/src/main/resources/lib/layout/breadcrumbs.css
Comment (aakashkanvinde) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: Yes, I totally agree with that. I will clear the new line and edit it directly in the Github. Should I make a new PR or just edit it in this one only?

**Comment by aakashkanvinde** [comment]
> The build failure looks unrelated, but just in case I would trigger a rebuild.
> 
> Close the pr, wait like 30 seconds to a minute, then re-open the PR, it should trigger a rebuild that way.
> 
> ✔ help reviewers know its ready

Yes, I have seen this thing happening multiple times. The build sometimes proceeds and checks out just fine, but then it rebuilds again and fails for unknown reasons.

File: core/src/main/resources/lib/layout/breadcrumbs.css
Comment (halkeye) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: You don't need a new PR. Its still related to this change. Edit your branch (ui or whatever) and it'll get picked up.

**Comment by halkeye** [comment]
> Yes, I have seen this thing happening multiple times. The build sometimes proceeds and checks out just fine, but then it rebuilds again and fails for unknown reasons.

The jenkins infra is so huge, there's so many things happening at the same time. They are constantly improving it, but sadly its still a free time type project, so failures happen.

File: core/src/main/resources/lib/layout/breadcrumbs.css
Comment (aakashkanvinde) [reply core/src/main/resources/lib/layout/breadcrumbs.css]: Sounds great! I will do that

**Comment by oleg-nenashev** [comment]
Will merge tomorrow if no negative feedback

**Comment by oleg-nenashev** [comment]
The build itself has passed. Looks like there was some concurrency issue on ACI `[2019-08-16T07:25:45.336Z] ERROR: missing workspace /home/jenkins/workspace/Core_jenkins_PR-4121 on aci-maven-11-6rjrq`. Will retrigger the build

**Comment by aakashkanvinde** [comment]
> The build itself has passed. Looks like there was some concurrency issue on ACI `[2019-08-16T07:25:45.336Z] ERROR: missing workspace /home/jenkins/workspace/Core_jenkins_PR-4121 on aci-maven-11-6rjrq`. Will retrigger the build

so will the commits be merged after the retriggered build passes?

**Comment by oleg-nenashev** [comment]
Yes, it will

**Comment by aakashkanvinde** [comment]
> Yes, it will

Super, thanks!

**Comment by oleg-nenashev** [comment]
Thanks for your contribution @aakashkanvinde !

**Comment by aakashkanvinde** [comment]
> Thanks for your contribution @aakashkanvinde !

@oleg-nenashev just helping the community 💯

**Pull request merged**
